functions/go/set-image/transformer/images.go

```diff
@@ -0,0 +1,237 @@
+package transformer
```

Comment:
main.go
```go
	if err := fn.AsMain(&SetImage{}); err != nil {
```
transformers/setImage.go
```go

var _ fn.Runner = &SetImage{}

type SetImage struct {
	 // DEPRECATED
        Name string `yam: "", json:"name,omitempty"`
	NewImage string `json:"newImage,omitempty"`
	NewTag string `json:"newTag,omitempty"`
        // ONLY for kustomize
        //additionalFieldSpec is private attribute, different than other config because if importing SetImage, we do not want users to use additionalFieldSpec. 
        additionalFieldSpec ....  
}

func (r *SetLabels) Run(ctx *fn.Context, _ *fn.KubeObject, items fn.KubeObjects) {
        err := r.Validate()
        if err != nil { 
               ctx.ResultErrorOrDie()
        }
        for _, o := range items.Where(hasPodSpec){
                    err := o.replaceImage()
                    if err != nil {
                         ctx.ResultError(err.Error(), o)
                    } else {
                          ctx.ResultInfoXXX("", o)
                    }
        }
          
        for _, o := range items.Where(hasPodTemplateSpec){
               ...
        }
         if lr.additionalFieldSpec != nil && len(r.additionalFieldSpec) != 0 {
               results := custom.SetAdditionalFieldSpec
               // converts results to results
               // log results
         } 
        ctx.ResultInfo("summary")
}
```
custom/addadditionalfield
```go

```


///